src/bin/ion/commands/generate/templates/java/class.templ

```diff
@@ -66,7 +100,7 @@ import java.io.IOException;
                 {% set field_occurrence = field_val.1 %}
                 {% if field_occurrence == "Optional" %} {% set field_value = field_value | primitive_data_type %} {% endif %}
                 case "{{ field_name }}":
-                    {{ field_name | camel }} =  java.util.Optional.of(
+                  {{ model.name | camel }}Builder = {{ model.name | camel }}Builder.{{ field_name | camel }}(
```

Comment:
The builder is borrowed mutably and doesn't require you do anything with the returned value, so we can drop the assignment. With the hard-coded name suggested above, it becomes simpler.
```suggestion
                  builder.{{ field_name | camel }}(
```